### DISCUSSION TITLE
Visualising timer timestamp on a video

### DISCUSSION THREAD

**Discussion by MilanNedoma** [bonsai-rx/bonsai#1580] [General]
Hi!

Just a quick question. Is there a simple way in Bonsai to get to create a timer with a timestamp to show up on a video in the "Image" node, without writing a script externally?

Thank you very much in advance.

**Comment by bruno-f-cruz** [comment]
Is this for online visualization or just saving an embedded string of text?

**Reply by MilanNedoma** [reply]
It would be probably the online visualization.
Essentially, I need to ensure that whilst the recording is on, I can have a way to precisely tell how much time has elapsed. 

I do apologize if I interpreted your question incorrectly.

**Reply by bruno-f-cruz** [reply]
For online, i would advise you against composing text onto images, as this is a relatively expensive process. Instead, you can use the following logic to make a stopwatch:

![image](https://github.com/bonsai-rx/bonsai/assets/7049351/a0c4a3ef-9601-45dd-b60b-3117383689d2)

The `Period` property in `Timer` defines the refresh rate of the timer.

**Reply by MilanNedoma** [reply]
May I just ask what the Subtract at the end is specifically?
Because I can´t seem to change it´s properties and by default, it does not allow to be placed for timestamps, as it is a binary operator?

**Reply by bruno-f-cruz** [reply]
Are you sure your workflow is identical to what I sent? I am attaching the code just in case:

<details>
    <summary>Workflow (Copy+Paste me)</summary>
    
```xml

<?xml version="1.0" encoding="utf-8"?>
<WorkflowBuilder Version="2.8.1"
                 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                 xmlns:rx="clr-namespace:Bonsai.Reactive;assembly=Bonsai.Core"
                 xmlns="https://bonsai-rx.org/2018/workflow">
  <Workflow>
    <Nodes>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="rx:Timer">
          <rx:DueTime>PT0S</rx:DueTime>
          <rx:Period>PT0S</rx:Period>
        </Combinator>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="rx:Timestamp" />
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="rx:Take">
          <rx:Count>1</rx:Count>
        </Combinator>
      </Expression>
      <Expression xsi:type="Combinator">
        <Combinator xsi:type="rx:CombineLatest" />
      </Expression>
      <Expression xsi:type="MemberSelector">
        <Selector>Item1.Timestamp,Item2.Timestamp</Selector>
      </Expression>
      <Expression xsi:type="Subtract" />
    </Nodes>
    <Edges>
      <Edge From="0" To="1" Label="Source1" />
      <Edge From="1" To="2" Label="Source1" />
      <Edge From="1" To="3" Label="Source1" />
      <Edge From="2" To="3" Label="Source2" />
      <Edge From="3" To="4" Label="Source1" />
      <Edge From="4" To="5" Label="Source1" />
    </Edges>
  </Workflow>
</WorkflowBuilder>

```

</details>

**Reply by MilanNedoma** [reply]
Thank you very much for this.

I have taken a look at the workflow and I realized my mistake.
Apologies for the delay in reply, everything works now!
